### PR TITLE
Feature/iojs constructors

### LINKED ISSUE #16: Support iojs stream simplified constructors
[As seen here.](https://iojs.org/api/stream.html#stream_simplified_constructor_api)

### PR DESCRIPTION
**Pull request by crowelch**
All tests passing, resolves #16

### CODE REVIEW

File: test/convenience-constructor-test.js [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+		});
+	});
+});
+
+describe('Duplex._flush', function () {
+	describe('data mode', function() {
+		it('should verify flush is called', function(done) {
+			var flushed = false;
+			var duplexStream = new Duplex({
+				write: function(chunk, encoding, cb) {
```
Comment (oppenlander) [inline test/convenience-constructor-test.js]: Add a `read` method to this as well. It should be called once.

Comment (oppenlander) [inline test/convenience-constructor-test.js]: Or rather, combine the read/write trests. A duplex should implement both, especially if piped through a readable/writable.

File: README.md [hunk 2]
```diff
@@ -20,6 +20,36 @@ var Writable = zstreams.Writable,
 The base zstreams function like, and inherit from, the native Node.JS streams.  They mostly behave like their
 parents, with
 
+### Simplified Constructors
+
+Zstreams supports iojs simplified stream constructors as seen [here](https://iojs.org/api/stream.html#stream_simplified_constructor_api).
+
+Additionally the `_flush` method has been added to writable and duplex streams.
```
Comment (crispy1989) [inline README.md]: This line doesn't belong in this section of the readme, and should probably be moved.

File: README.md [hunk 3]
```diff
@@ -20,6 +20,36 @@ var Writable = zstreams.Writable,
 The base zstreams function like, and inherit from, the native Node.JS streams.  They mostly behave like their
 parents, with
 
+### Simplified Constructors
```
Comment (crispy1989) [inline README.md]: This whole section should probably be moved down.  A readme should be organized according to how pertinent the information is to typical use cases.  Someone is a lot more likely to need to convert a native stream than to use a simplified constructor format (for example).

File: test/convenience-constructor-test.js [hunk 4]
```diff
@@ -0,0 +1,356 @@
...
+				}
+			});
+			var count = 0;
+			zstreams.fromString('abcd', {chunkSize: 1})
+				.pipe(writeStream)
+				.intoCallback(function(error) {
+					expect(error).to.not.exist;
+					expect(count).to.equal(4);
+					done();
+				});
```
Comment (crispy1989) [inline test/convenience-constructor-test.js]: Weird indentation?

File: test/convenience-constructor-test.js [hunk 5]
```diff
@@ -0,0 +1,356 @@
...
```
Comment (crispy1989) [inline test/convenience-constructor-test.js]: Weird indentation?

File: test/convenience-constructor-test.js [hunk 6]
```diff
@@ -0,0 +1,356 @@
...
+					cb();
+				}
+			});
+			zstreams.fromString('abcd')
+				.pipe(writeStream)
+				.intoCallback(function(error) {
+					expect(error).to.not.exist;
+					expect(flushed).to.be.true;
+					done();
+				});
```
Comment (crispy1989) [inline test/convenience-constructor-test.js]: Weird indentation?

File: test/convenience-constructor-test.js [hunk 7]
```diff
@@ -0,0 +1,356 @@
...
```
Comment (crispy1989) [inline test/convenience-constructor-test.js]: Weird indentation?

Comment (crispy1989) [inline test/convenience-constructor-test.js]: There's weird indentation throughout this, please fix so there aren't multiple indentation step differences.

**Pull request merged**